crates/bitwarden-crypto/src/xchacha20.rs

```diff
@@ -22,12 +22,12 @@ use crate::CryptoError;
 
 #[allow(unused)]
 pub(crate) struct XChaCha20Poly1305Ciphertext {
-    nonce: GenericArray<u8, <XChaCha20Poly1305 as AeadCore>::NonceSize>,
-    ciphertext: Vec<u8>,
+    pub(crate) nonce: GenericArray<u8, <XChaCha20Poly1305 as AeadCore>::NonceSize>,
+    pub(crate) ciphertext: Vec<u8>,
```

Comment:
why does this need to be public?

I think you're pretty set on building the cose message outside of these modules, but even so readers should be preferred to keep them immutable

Reply:
Yes, I think they're better represented as getters. Not sure why I said readers, brain fart I guess. 

The point is that you _don't_ want them public because then if someone marks your struct as `mut`, all of a sudden your nonce and ciphertext are `mut` as well.